src/lib/yaraparse.py

```diff
@@ -108,42 +108,49 @@ def str_expr(condition, rule_strings) -> Optional[str]:
     return ursify_string(rule_strings[condition.id])
 
 
-def str_wild_expr(condition, rule_strings) -> Optional[str]:
+
+def expand_string_wildcard(condition, rule_strings) -> List[str]:
     condition_regex = re.escape(condition.text)
     condition_regex = condition_regex.replace("\\*", ".*")
     filtered_strings = [
         v for k, v in rule_strings.items() if re.match(condition_regex, k)
     ]
 
     ursa_strings = [ursify_string(x) for x in filtered_strings]
-    strings = [s for s in ursa_strings if s is not None]
+    return [s for s in ursa_strings if s is not None]
 
-    if strings:
-        return ", ".join(strings)
-    return None
+
+def expand_set_expression(children: SetExpression, rule_strings) -> List[str]:
+    parsed_elements = []
+    for e in children.elements:
+        if type(e) is StringWildcardExpression:
+            parsed_elements += expand_string_wildcard(e, rule_strings)
+        elif type(e) is StringExpression:
+            parsed_elements.append(str_expr(e, rule_strings))
+        else:
+            print("WTF", type(e))
```

Comment:
I guess there's a nicer way to do this

Reply:
:thinking: don't even think about touching my debug prints